### ISSUE TITLE
Save progress: Manual accessibility testing

### ISSUE THREAD

**Issue by srtalbot** [cds-snc/platform-forms-client#5089]
### Acceptance criteria

- [x] UI supports 400% magnification. 
- [x] Focus states are highlighted when navigating by keyboard.
- [x] Screenreader users know about the “save” and “resume” buttons, but they do not interrupt the flow of completing the form. 
- [x] Download successes and failures are announced
  - [x] Toast error message fails to announce. 
- [ ] Error messages are descriptive enough to recover
  - [ ] Error message is too generic
- [x] Re-upload announces the file selection (covered by OS)
- [x] Re-upload announces changes in the file selection (covered by OS)
- [x] Re-upload announces the file that is selected and uploaded (covered by OS)
- [x] State changes are announced so the user knows which page they are on.
- [x] Provide findings and methodology to the Design System team. 

**Issues found**
- [x] When we use the back or edit buttons, the focus is not returned
- [x] When a user closes a dialogue, then the focus is not returned to a specific place

**Comment by timarney** [comment]
@thiessenp-cds we'll likely want to adjust the mobile buttons for next / back where we have "just icons" ... we'll likely want to move the title up to the button level vs the svg level.

**Comment by thiessenp-cds** [comment]
Note that 200% meets the requirements (WCAG 2 AA). The design also scales to 400% which is a nice bonus.

![Image](https://github.com/user-attachments/assets/caa63bc8-5f35-4eed-a05d-21436f1d3a64)

**Comment by thiessenp-cds** [comment]
I think I need @anikbrazeau's help with this one. Would you say the messages are descriptive for: "Error messages are descriptive enough to recover"

There is the case where the data is bad in a download file. When this happens there is a generic error message in a toast. In this case there is not really a way to recover so I think this is the best we can do?

**Comment by thiessenp-cds** [comment]
Here is a quick summary of the minor issues I found from testing. Most (/all) of the issues existed before Save Progress.

- (unrelated) skip-to link needs to be fixed
- Response list DT/DT missing a parent DL
- Clicking Go Back navigates to the previous page but focus appears lost 
- Save dialog when clicking cancel or save and returning to the app, focus appears to be lost
- Error toast from a file with invalid data, the error message is not announced

**Comment by timarney** [comment]
For Toast we can look to do the update https://github.com/cds-snc/platform-forms-client/issues/5143 to fix.

**Comment by anikbrazeau** [comment]
@thiessenp-cds 

We can definitely try and be more descriptive to help people recover -- let me see what I can do 
(for now it might just be try again / restart which is not ideal). 

In future, having a support avenue should help here for a better path to recovery.

**Comment by thiessenp-cds** [comment]
@samsadasivan put me in touch with Nick from the design team and shared briefly the methodology I used to test this feature.

**Comment by thiessenp-cds** [comment]
From what I understand we'll leave the current error message as is. 
If we want to update the error message to be more descriptive, perhaps we could open another ticket. I'll leave it unchecked though as a reminder that we decided to leave it as is but may want to come back to it in the future.
